azure-ts-appservice-devops/azure-pipelines.yml

```diff
@@ -0,0 +1,26 @@
+jobs:
+- job: infrastructure
+  pool:
+    vmImage: 'ubuntu-16.04'
+  steps:
+  - task: DotNetCoreCLI@2
+    inputs:
+      command: 'publish'
+      projects: 'src'
+      zipAfterPublish: false
+      modifyOutputPath: false
+    displayName: 'Build and publish ASP.NET Core app'
+  - script: |
```

Comment:
Yeah it would be great to use the task extension available in public preview here: https://marketplace.visualstudio.com/items?itemName=pulumi.build-and-release-task.

You could then convert this `script` task to ditch the bash scripts:

```yaml
- task: Pulumi@0
    inputs:
      azureSubscription: "Service Connection to your Azure Subscription"
      command: "up"
      args: "--yes"
      cwd: "pulumi-stack/"
      stack: "dev"

# At this point the pulumi tool should be in the path of the agent. So you should be able to use it in a subsequent step, but within the same job.
- script: |
    # Set config properties
     pulumi config set --secret sqlUsername ${SQL_USERNAME}
     pulumi config set --secret sqlPassword ${SQL_PASSWORD}

# Or, you can also use the Pulumi task once again. This may be a little verbose than just running an inline script like above if you are wanting to run `config set...` commands.
- Pulumi@0
   inputs:
     azureSubscription: "Service Connection to your Azure Subscription"
     command: "config"
     args: "set --secret sqlUserName $(sql.username)"
     cwd: "pulumi-stack/"
     stack: "dev"

- Pulumi@0
   inputs:
     azureSubscription: "Service Connection to your Azure Subscription"
     command: "config"
     args: "set --secret sqlPassword $(sql.password)"
     cwd: "pulumi-stack/"
     stack: "dev"
```

@mikhailshilkov would you mind installing the extension to your organization and giving it a try with your example here? I would love your feedback on it! Afterwards, it would be great to include an example of the pipeline config that shows how to use the task extension as well in addition to the current one that you have here.

I think we should still include merge this example in its current form, since not everyone may have access to installing extensions into their organization, and as such I don't want folks trying the example to be blocked by not being able to install the extension.

Reply:
Done! Thanks for the task, it simplifies the setup a lot!